[BUG] Extendable MMUs with more than 8 slots fail with a compile error from steppers.h

### Did you test the latest `bugfix-2.0.x` code?

Yes, and the problem still exists.

### Bug Description

I just upgraded Marlin to 2.0.9.3 and found that it no longer compiles with the MMU enabled and EXTRUDERS set to > 8, which should be supported.

```
Marlin/src/feature/../module/stepper.h:249:1: error: macro "LIST_15" requires 16 arguments, but only 8 given
         bool LIST_N(EXTRUDERS, E0:1, E1:1, E2:1, E3:1, E4:1, E5:1, E6:1, E7:1);
```

### Bug Timeline

Wasn't present in 2.0.9.1, is present in 2.0.9.3

### Expected behavior

I should be able to compile Marlin.

### Actual behavior

I can't compile Marlin.

### Steps to Reproduce

1. Set `#define MMU_MODEL EXTENDABLE_EMU_MMU2S` in `Configuration.h` (and enable dependencies `FILAMENT_RUNOUT_SENSOR`, `NOZZLE_PARK_FEATURE` and `EMERGENCY_PARSER`)
2. Set `#define EXTRUDERS 10` in `Configuration.h`
3. Try to compile, and observe the reported compile error.

### Version of Marlin Firmware

2.0.9.3

### Printer model

_No response_

### Electronics

_No response_

### Add-ons

_No response_

### Bed Leveling

_No response_

### Your Slicer

_No response_

### Host Software

_No response_

### Additional information & file uploads

It looks like the commit that caused this problem is 021ceeba0b0ccadd7246d5e2da56df7868349206. There might need to be a special case there handling setups with an extendable MMU, since they only have one extruder motor, even if EXTRUDERS is set to more than 1.

Also, this is the second time in a year that i noticed that a commit in Marlin broke compilation with extendable MMUs which indicates that it's easy to overlook, it might be worth considering adding a test to avoid this from happening in the future.

--- a/Marlin/src/MarlinCore.cpp
+++ b/Marlin/src/MarlinCore.cpp
@@ -676,28 +676,9 @@ inline void manage_inactivity(const bool no_stepper_sleep=false) {
       && ELAPSED(ms, gcode.previous_move_ms + SEC_TO_MS(EXTRUDER_RUNOUT_SECONDS))
       && !planner.has_blocks_queued()
     ) {
-      #if HAS_SWITCHING_EXTRUDER
-        bool oldstatus;
-        switch (active_extruder) {
-          default: oldstatus = stepper.AXIS_IS_ENABLED(E_AXIS, 0); stepper.ENABLE_EXTRUDER(0); break;
-          #if E_STEPPERS > 1
-            case 2: case 3: oldstatus = stepper.AXIS_IS_ENABLED(E_AXIS, 1); stepper.ENABLE_EXTRUDER(1); break;
-            #if E_STEPPERS > 2
-              case 4: case 5: oldstatus = stepper.AXIS_IS_ENABLED(E_AXIS, 2); stepper.ENABLE_EXTRUDER(2); break;
-              #if E_STEPPERS > 3
-                case 6: case 7: oldstatus = stepper.AXIS_IS_ENABLED(E_AXIS, 3); stepper.ENABLE_EXTRUDER(3); break;
-              #endif // E_STEPPERS > 3
-            #endif // E_STEPPERS > 2
-          #endif // E_STEPPERS > 1
-        }
-      #else // !HAS_SWITCHING_EXTRUDER
-        bool oldstatus;
-        switch (active_extruder) {
-          default:
-          #define _CASE_EN(N) case N: oldstatus = stepper.AXIS_IS_ENABLED(E_AXIS, N); stepper.ENABLE_EXTRUDER(N); break;
-          REPEAT(E_STEPPERS, _CASE_EN);
-        }
-      #endif
+      const int8_t e_stepper = TERN(HAS_SWITCHING_EXTRUDER, active_extruder >> 1, active_extruder);
+      const bool e_off = !stepper.AXIS_IS_ENABLED(E_AXIS, e_stepper);
+      if (e_off) stepper.ENABLE_EXTRUDER(e_stepper);
 
       const float olde = current_position.e;
       current_position.e += EXTRUDER_RUNOUT_EXTRUDE;
@@ -706,22 +687,7 @@ inline void manage_inactivity(const bool no_stepper_sleep=false) {
       planner.set_e_position_mm(olde);
       planner.synchronize();
 
-      #if HAS_SWITCHING_EXTRUDER
-        switch (active_extruder) {
-          default: if (oldstatus) stepper.ENABLE_EXTRUDER(0); else stepper.DISABLE_EXTRUDER(0); break;
-          #if E_STEPPERS > 1
-            case 2: case 3: if (oldstatus) stepper.ENABLE_EXTRUDER(1); else stepper.DISABLE_EXTRUDER(1); break;
-            #if E_STEPPERS > 2
-              case 4: case 5: if (oldstatus) stepper.ENABLE_EXTRUDER(2); else stepper.DISABLE_EXTRUDER(2); break;
-            #endif // E_STEPPERS > 2
-          #endif // E_STEPPERS > 1
-        }
-      #else // !HAS_SWITCHING_EXTRUDER
-        switch (active_extruder) {
-          #define _CASE_RESTORE(N) case N: if (oldstatus) stepper.ENABLE_EXTRUDER(N); else stepper.DISABLE_EXTRUDER(N); break;
-          REPEAT(E_STEPPERS, _CASE_RESTORE);
-        }
-      #endif // !HAS_SWITCHING_EXTRUDER
+      if (e_off) stepper.DISABLE_EXTRUDER(e_stepper);
 
       gcode.reset_stepper_timeout(ms);
     }

--- a/Marlin/src/gcode/control/M17_M18_M84.cpp
+++ b/Marlin/src/gcode/control/M17_M18_M84.cpp
@@ -71,7 +71,7 @@ void do_enable(const stepper_flags_t to_enable) {
 
   if (!shall_enable) return;    // All specified axes already enabled?
 
-  ena_mask_t also_enabled = 0;    // Track steppers enabled due to overlap
+  ena_mask_t also_enabled = 0;  // Track steppers enabled due to overlap
 
   // Enable all flagged axes
   LOOP_NUM_AXES(a) {

--- a/Marlin/src/module/stepper.h
+++ b/Marlin/src/module/stepper.h
@@ -266,7 +266,7 @@ typedef struct {
     struct {
       bool NUM_AXIS_LIST(X:1, Y:1, Z:1, I:1, J:1, K:1, U:1, V:1, W:1);
       #if HAS_EXTRUDERS
-        bool LIST_N(EXTRUDERS, E0:1, E1:1, E2:1, E3:1, E4:1, E5:1, E6:1, E7:1);
+        bool LIST_N(E_STEPPERS, E0:1, E1:1, E2:1, E3:1, E4:1, E5:1, E6:1, E7:1);
       #endif
     };
   };

--- a/Marlin/src/module/stepper.h
+++ b/Marlin/src/module/stepper.h
@@ -265,7 +265,7 @@ typedef struct {
     ena_mask_t bits;
     struct {
       bool NUM_AXIS_LIST(X:1, Y:1, Z:1, I:1, J:1, K:1, U:1, V:1, W:1);
-      #if HAS_EXTRUDERS
+      #if E_STEPPERS
         bool LIST_N(E_STEPPERS, E0:1, E1:1, E2:1, E3:1, E4:1, E5:1, E6:1, E7:1);
       #endif
     };

--- a/Marlin/src/module/stepper.h
+++ b/Marlin/src/module/stepper.h
@@ -267,7 +267,7 @@
 #endif
 
 // Number of axes that could be enabled/disabled. Dual/multiple steppers are combined.
-#define ENABLE_COUNT (NUM_AXES + E_STEPPERS)
+#define ENABLE_COUNT (NUM_AXES + E_STATES)
 typedef bits_t(ENABLE_COUNT) ena_mask_t;
 
 // Axis flags type, for enabled state or other simple state
@@ -276,8 +276,8 @@ typedef struct {
     ena_mask_t bits;
     struct {
       bool NUM_AXIS_LIST(X:1, Y:1, Z:1, I:1, J:1, K:1, U:1, V:1, W:1);
-      #if E_STEPPERS
-        bool LIST_N(E_STEPPERS, E0:1, E1:1, E2:1, E3:1, E4:1, E5:1, E6:1, E7:1);
+      #if E_STATES
+        bool LIST_N(E_STATES, E0:1, E1:1, E2:1, E3:1, E4:1, E5:1, E6:1, E7:1);
       #endif
     };
   };

--- a/Marlin/src/module/stepper.h
+++ b/Marlin/src/module/stepper.h
@@ -255,6 +255,17 @@
 // This does not account for the possibility of multi-stepping.
 #define MIN_STEP_ISR_FREQUENCY (MAX_STEP_ISR_FREQUENCY_1X >> 1)
 
+// TODO: Review and ensure proper handling for special E axes with commands like M17/M18, stepper timeout, etc.
+#if ENABLED(MIXING_EXTRUDER)
+  #define E_STATES EXTRUDERS  // All steppers are set together for each mixer. (Currently limited to 1.)
+#elif HAS_SWITCHING_EXTRUDER
+  #define E_STATES E_STEPPERS // One stepper for every two EXTRUDERS. The last extruder can be non-switching.
+#elif HAS_PRUSA_MMU2
+  #define E_STATES E_STEPPERS // One E stepper shared with all EXTRUDERS, so setting any only sets one.
+#else
+  #define E_STATES E_STEPPERS // One stepper for each extruder, so each can be disabled individually.
+#endif
+
 // Number of axes that could be enabled/disabled. Dual/multiple steppers are combined.
 #define ENABLE_COUNT (NUM_AXES + E_STEPPERS)
 typedef bits_t(ENABLE_COUNT) ena_mask_t;

